WooCommerce/Classes/ViewRelated/Editor/AztecEditorViewController.swift

```diff
@@ -0,0 +1,863 @@
+    }
+
+    @objc func applicationWillResignActive(_ notification: Foundation.Notification) {
+
+        // [2018-03-05] Need to close the options VC on backgrounding to prevent view hierarchy inconsistency crasher.
+        optionsTablePresenter.dismiss()
+    }
+}
+
+// MARK: - Navigation actions
+//
+private extension AztecEditorViewController {
+    @objc func cancelButtonTapped() {
+        navigationController?.popViewController(animated: true)
+    }
+
+    @objc func saveButtonTapped() {
+        let content = getHTML()
+        onContentSave?(content)
+
+        navigationController?.popViewController(animated: true)
+    }
+}
+
+// MARK: - TextViewAttachmentDelegate Conformance
+//
+extension AztecEditorViewController: TextViewAttachmentDelegate {
+    func textView(_ textView: TextView, attachment: NSTextAttachment, imageAt url: URL, onSuccess success: @escaping (UIImage) -> Void, onFailure failure: @escaping () -> Void) {
+
+    }
+
+    func textView(_ textView: TextView, urlFor imageAttachment: ImageAttachment) -> URL? {
+        return nil
+    }
+
+    func textView(_ textView: TextView, placeholderFor attachment: NSTextAttachment) -> UIImage {
+        return UIImage.cameraImage
+    }
+
+    func textView(_ textView: TextView, deletedAttachment attachment: MediaAttachment) {
+
+    }
+
+    func textView(_ textView: TextView, selected attachment: NSTextAttachment, atPosition position: CGPoint) {
+
+    }
+
+    func textView(_ textView: TextView, deselected attachment: NSTextAttachment, atPosition position: CGPoint) {
+
+    }
+}
+
+private extension AztecEditorViewController {
+    func updateFormatBar() {
+        switch editorView.editingMode {
+        case .html:
+            updateFormatBarForHTMLMode()
+        case .richText:
+            updateFormatBarForVisualMode()
+        }
+    }
+
+    /// Updates the format bar for HTML mode.
+    ///
+    func updateFormatBarForHTMLMode() {
+        assert(editorView.editingMode == .html)
+
+        guard let toolbar = richTextView.inputAccessoryView as? Aztec.FormatBar else {
+            return
+        }
+
+        toolbar.selectItemsMatchingIdentifiers([FormattingIdentifier.sourcecode.rawValue])
+    }
+
+    /// Updates the format bar for visual mode.
+    ///
+    func updateFormatBarForVisualMode() {
+        assert(editorView.editingMode == .richText)
+
+        guard let toolbar = richTextView.inputAccessoryView as? Aztec.FormatBar else {
+            return
+        }
+
+        var identifiers = Set<FormattingIdentifier>()
+
+        if richTextView.selectedRange.length > 0 {
+            identifiers = richTextView.formattingIdentifiersSpanningRange(richTextView.selectedRange)
+        } else {
+            identifiers = richTextView.formattingIdentifiersForTypingAttributes()
+        }
+
+        toolbar.selectItemsMatchingIdentifiers(identifiers.map({ $0.rawValue }))
+    }
+}
+
+// MARK: - UITextViewDelegate methods
```

Comment:
It seems like all actions in this extension also end up touching the format bar somehow, wo maybe these could be extracted as well?